worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java

```diff
@@ -346,11 +350,32 @@ public void load(
                 }
                 UUID uuid = UUID.fromString(filename.substring(4));
                 URL webUrl = new URL(Settings.settings().WEB.URL);
-                format = ClipboardFormats.findByAlias(formatName);
+                if ((format = ClipboardFormats.findByAlias(formatName)) == null) {
+                    actor.print(Caption.of("worldedit.schematic.unknown-format", TextComponent.of(formatName)));
+                    return;
+                }
+                // The interface requires the correct schematic extension - otherwise it can't be downloaded
+                // So it basically only supports .schem files (sponge v2 + v3) - or the correct extensions is specified manually
+                // Sadly it's not really an API endpoint but spits out the HTML source of the uploader - so no real handling
+                // can happen
                 URL url = new URL(webUrl, "uploads/" + uuid + "." + format.getPrimaryFileExtension());
-                ReadableByteChannel byteChannel = Channels.newChannel(url.openStream());
-                in = Channels.newInputStream(byteChannel);
-                uri = url.toURI();
+                final Path temp = Files.createTempFile("faweremoteschem", null);
+                final File tempFile = temp.toFile();
+                // delete temporary file when we're done
+                closer.register((Closeable) () -> Files.deleteIfExists(temp));
+                // write schematic into temporary file
+                try (final ReadableByteChannel byteChannel = Channels.newChannel(url.openStream());
+                     final FileOutputStream out = new FileOutputStream(tempFile);
+                     final FileChannel outChannel = out.getChannel()) {
+                    outChannel.transferFrom(byteChannel, 0, Long.MAX_VALUE);
```

Comment:
`InputStream` also has a `transferTo` method, I think we could skip channels (and probably `File`?) here

Reply:
The file is still needed - once for actually having an InputStream, and for `ClipboardFormats.findByFile` which requires the file